tasks/build/create_build.js

```diff
@@ -1,84 +1,52 @@
 
+var rewritePackage = require('./rewrite_package');
+var gitInfo = require('./git_info');
+
 module.exports = function createBuild(plugin, buildVersion, kibanaVersion, files) {
-  var buildId = `${plugin.id}-${buildVersion}`;
   var buildSource = plugin.root;
   var buildTarget = join(plugin.root, 'build');
-
-  return new Promise(function (resolve) {
-    vfs
-      .src(files, { cwd: buildSource, base: buildSource })
-
-      // modify the package.json file
-      .pipe(map(function (file) {
-        if (file.basename === 'package.json' && file.dirname === buildSource) {
-          var pkg = JSON.parse(file.contents.toString('utf8'));
-
-          // rewrite the target kibana version while the
-          // file is on it's way to the archive
-          if (!pkg.kibana) pkg.kibana = {};
-          pkg.kibana.version = kibanaVersion;
-          pkg.version = buildVersion;
-
-          // append build info
-          pkg.build = {
-            git: gitInfo(buildSource),
-            date: new Date().toString()
-          };
-
-          file.contents = toBuffer(JSON.stringify(pkg, null, 2));
-        }
-
-        return file;
-      }))
-
-      // put all files inside the correct directories
-      .pipe(rename(function nestFileInDir(path) {
-        var nonRelativeDirname = path.dirname.replace(/^(\.\.\/?)+/g, '');
-        path.dirname = join('kibana', plugin.id, nonRelativeDirname);
-      }))
-
-      .pipe(zip(`${buildId}.zip`))
-      .pipe(vfs.dest(buildTarget))
-      .on('end', resolve);
-  });
-};
-
-function toBuffer(string) {
-  if (typeof Buffer.from === 'function') {
-    return Buffer.from(string, 'utf8');
-  } else {
-    // this was deprecated in node v5 in favor
-    // of Buffer.from(string, encoding)
-    return new Buffer(string, 'utf8');
-  }
-}
-
-function gitInfo(rootPath) {
-  try {
-    var LOG_SEPARATOR = '||';
-    var commitCount = execFileSync('git', ['rev-list', '--count', 'HEAD'], {
-      cwd: rootPath,
-      stdio: ['ignore', 'pipe', 'ignore'],
-      encoding: 'utf8',
+  var buildRoot = join(buildTarget, 'kibana', plugin.id);
+
+  return del(buildTarget)
+    .then(function () {
+      return new Promise(function (resolve, reject) {
+        vfs
+          .src(files, { cwd: buildSource, base: buildSource })
+          // modify the package.json file
+          .pipe(rewritePackage(buildSource, buildVersion, kibanaVersion))
+
+          // put all files inside the correct directories
+          .pipe(rename(function nestFileInDir(path) {
+            var nonRelativeDirname = path.dirname.replace(/^(\.\.\/?)+/g, '');
+            path.dirname = join(relative(buildTarget, buildRoot), nonRelativeDirname);
+          }))
+
+          .pipe(vfs.dest(buildTarget))
+          .on('end', resolve)
+          .on('error', reject);
+      });
+    })
+    .then(function () {
+      // install packages in build
+      var options = {
+        cwd: buildRoot,
+        stdio: ['ignore', 'ignore', 'pipe'],
```

Comment:
I ***think*** that npm uses stderr to print errors specifically, so this probably accomplished that goal.

I'm not suggesting that there is particularly useful information in there, but I think it's better to not modify the output of the command in case there is something unexpectedly useful, and also to help indicate progress.